api/src/main/java/jakarta/data/TextRestrictionRecord.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data;
+
+// Internal implementation class.
+// The proper way for users to obtain instances is via
+// the static metamodel or Restrict.* methods 
+
+record TextRestrictionRecord<T>(
```

Comment:
You can use `DefaultTextRestriction`

Reply:
I suppose another idea would be `TextRestrictionImpl`. That would make it very clear that it is a hidden internal class used for implementation.

Reply:
I am OK with either `...Record`or `Default`, but I would avoid the `TextRestrictionImpl`

Reply:
Alright. `Default` seemed awkward to me, so I'll leave it as `TextRestrictionRecord` for now. We can rename it later if we think of something better.